src/story.ts

```diff
@@ -356,6 +363,16 @@ export type BaseAnnotations<TRenderer extends Renderer = Renderer, TArgs = Args>
    */
   loaders?: LoaderFunction<TRenderer, TArgs>[] | LoaderFunction<TRenderer, TArgs>;
 
+  /**
+   * Function to be called before each story. When the function is async, it will be awaited.
+   *
+   * `beforeEach` can be added to preview, the default export and to a specific story.
+   * They are run (and awaited) in the order: preview, default export, story
```

Comment:
I think we should probably be more careful with our language here to avoid confusion:

- rather than "preview" -- I would say "at the project level" or similar
- rather than "the default export" -- I would say "component level"